### ISSUE TITLE
Crash at world startup with latest beta.

### ISSUE THREAD

**Issue by NikGold** [Azanor/thaumcraft-beta#858]
https://paste.dimdev.org/sohebozuni.txt

**Comment by Raumis** [comment]
I get the same problem
This kills the client and happens when you place nitor in the world.

**Comment by Grumpypeet** [comment]
I think I also had/have the same problem.  Client has been crashing after I look in the Thaumbook, then attempt to look through the Thaumeter.  Attached most recent crash report.
[crash-2018-05-24_11.08.00-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2036263/crash-2018-05-24_11.08.00-client.txt)

**Comment by UnixRonin** [comment]
I'm seeing the exact same problem.  I only have one nitor placed right now, in light of this post I'm going to roll back to beta 13 for just long enough to remove it and see if that resolves the problem.

**Comment by CTH999** [comment]
Same, just it's whenever I use Salis Mundus on a structer (Crash report coming up)

**Comment by CTH999** [comment]
[crash-2018-05-24_12.20.44-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2036459/crash-2018-05-24_12.20.44-client.txt)

Yes, I did have other mods, _but_ it had no problem with the other betas.

**Comment by UnixRonin** [comment]
OK, so, rolling back to beta 13 and removing my one placed nitor **did** let me join the server with beta 15.  But about five minutes later, I got the same client crash again without having placed anything at all anywhere.  So apparently it's not **just** nitors that cause the problem.

**Comment by enderger** [comment]
I can’t even get back on my smp lan world without crashing as soon as I hit resume

**Comment by drakray** [comment]
B15, https://pastebin.com/SQunwEns
```
java.lang.UnsupportedOperationException
	at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(Unknown Source)
	at thaumcraft.client.fx.ParticleEngine.updateParticles(ParticleEngine.java:296)
```
When looking through Thaumometer

**Comment by Matakor** [comment]
Adding my +1 to this. 
`java.lang.UnsupportedOperationException
	at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(Unknown Source)
	at thaumcraft.client.fx.ParticleEngine.updateParticles(ParticleEngine.java:296)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_224_ParticleEngine_updateParticles_ClientTickEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.fml.common.FMLCommonHandler.onPreClientTick(FMLCommonHandler.java:343)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1706)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1096)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)`

Render bug, I guess. Will be downgrading for now, hopefully the next update contains a fix.

**Comment by enderger** [comment]
Same as #860

**Comment by GeneralStukov** [comment]
Same as others, I'm on beta15 can post a pastebin of an error if required but others included theirs as well.

**Comment by MokahTGS** [comment]
https://paste.dimdev.org/oyanivubik.txt

My crash log.  This is a crash on loading a savegame.

**Comment by enderger** [comment]
Patches are needed, can I expect beta 16 TOMORROW because this could not have ever been tested at all.

**Comment by enderger** [comment]
And it breaks the game

**Comment by UnixRonin** [comment]
Um, dude, you know, **_it is a beta..._**  these things happen.

**Comment by MokahTGS** [comment]
Just roll back to beta 14

**Comment by enderger** [comment]
B14 also exhibits this bug for me

**Comment by enderger** [comment]
This must be a side effect of fixing the thaumometer lag spike, unpatching that bug shpuld flatten this bug

**Comment by enderger** [comment]
And my stabilizer only exists in b14

**Comment by yunruiwatts** [comment]
Looks like ["Element-changing operations on [`CopyOnWriteArrayList.COWIterator`s] (remove, set, and add) are not supported."](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CopyOnWriteArrayList.html)

**Comment by yunruiwatts** [comment]
Looks like #861, #865, #866, and #867 all gave the same stack traces as this one - trying to call `remove` on a `$COWIterator`

**Comment by enderger** [comment]
I’ll post my B14 crash(not going to be hard) when I can

**Comment by enderger** [comment]
I can not avoid this bug on B14, probably won’t work on B15

**Comment by enderger** [comment]
I believe that I know how the mod can be fixed and what has caused this mess: the patch for bug #478. To fix that bug, particles had to be edited, and there probably is a mistake in the new code. Undoing that fix probably will fix this bug. This bug is more serious than the thaumometer lag bug.

**Comment by MokahTGS** [comment]
How about we fix both bugs.  I'm not getting a crash with B14 and the Lagometer has been fixed.  Not sure what he did in B15, but it isn't affecting everyone.

**Comment by wamc2017** [comment]
Hmm, B14 should not crash with an error which mentions CopyOnWriteArrayList in thaumcraft.client.fx.ParticleEngine, because azanor used ArrayList before...

@enderger,
are you sure you're using B14 and got the same crash? If so upload your crashlog, because this can't be the exact same error, if the class used is different in both versions...

**Comment by enderger** [comment]
I am positive that I’m on b14: I updated before B15 even came out!

For those effected, are you using AMD graphics cards?

**Comment by DWX12** [comment]
Figure I'll add to this too. Looks like the same error on B15 and all I did was shift click a green crystal into my inventory. https://paste.dimdev.org/ukubucuyor.txt

**Comment by Sanofus** [comment]
I had the same severe issue. http://www.minecraftforge.net/forum/topic/64567-crash-unkown-source/

**Comment by UnixRonin** [comment]
nVidia GTX980, here, running 1080p.

**Comment by Adarcer** [comment]
I have had too stay at beta 13 the other 2 updates either would crash at startup due to better than mods compatibility not updated can be bypassed by disabling the comp module but either way you crash in game while doing nothing that would seem to trigger a crash. seemingly caused by the fx particle engine.. I suspect it has something to do with the wisps mobs that have begin to appear.  

https://pastebin.com/0NKzkN10

**Comment by enderger** [comment]
crash on B14: https://gist.github.com/enderger/c1ab4c7c40a9e6ae2c1f94fab1673b3a

**Comment by Sanofus** [comment]
I could solve the crash downgrading to beta 14. I happenend only in 15.

**Comment by enderger** [comment]
@Adarcer, any particle crashes thaumcraft in the new betas

**Comment by enderger** [comment]
Alright, so we can rule out AMD incompatibility(or NVidia incompatability).

**Comment by ChewedPencil** [comment]
yeah, does the same when you try and make foci

**Comment by enderger** [comment]
Regardless, I think that the bug existed in B14 but grew exponentially in B15

**Comment by yunruiwatts** [comment]
@enderger I could have told you that AMD vs NVidia didn't matter just from looking at the crash log. As for the bug growing, yeah that's one way to look at it - Azanor must've been trying to fix the bug you ran into, but the fix doesn't actually work.

For anyone curious about the technical details (it's fairly easy to tell just from looking at the crash logs), Azanor had been using an `ArrayList` (one option for using a list in Java) and ran into a `ConcurrentModificationException` - something changes the list while you're going through it in a way that means Java can't guarantee the state of the list at that point, so it throws the exception. Azanor tried to replace the `ArrayList` with a `CopyOnWriteArrayList`. I haven't used one of those before, but, based on the documentation for it, it looks like it is one way to get around needing to worry about concurrent modification. Unfortunately, the way that the code had been cleaning up the list doesn't work with the new kind of list.

**Comment by enderger** [comment]
I was looking for similarities between the effected computers

**Comment by enderger** [comment]
Time to get a bit dirty in eclipse!

**Comment by shelbykauth** [comment]
I got the same crash error with 
`java.lang.UnsupportedOperationException
	at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(CopyOnWriteArrayList.java:1176)
	at thaumcraft.client.fx.ParticleEngine.updateParticles(ParticleEngine.java:296)
        at ......`

I was in creative mode.  I do have a bunch of mods, but no other mod was mentioned in the stack trace.  I was not doing anything with Thaumcraft, but there was potentially a beehive with thaumic bees maybe, I don't know.

**Comment by xLastt** [comment]
Holy god i too

**Comment by jagereens** [comment]
pretty sure this is the issue i am experiencing myself at the moment i cannot figure out for the life of me what is causing it because i'll be simply exploring the world then hit a crash that continues to loop every time i try to start up the world again here is the crash report

---- Minecraft Crash Report -------- Minecraft Crash Report ----// Oops.
Time: 5/26/18 12:07 PMDescription: Unexpected error
java.lang.UnsupportedOperationException at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(CopyOnWriteArrayList.java:1176) at thaumcraft.client.fx.ParticleEngine.updateParticles(ParticleEngine.java:296) at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_59_ParticleEngine_updateParticles_ClientTickEvent.invoke(.dynamic) at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) at net.minecraftforge.fml.common.FMLCommonHandler.onPreClientTick(FMLCommonHandler.java:343) at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1706) at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1096) at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397) at net.minecraft.client.main.Main.main(SourceFile:123) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:483) at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

A detailed walkthrough of the error, its code path and all known details is as follows:---------------------------------------------------------------------------------------
-- Head --Thread: Client threadStacktrace: at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(CopyOnWriteArrayList.java:1176) at thaumcraft.client.fx.ParticleEngine.updateParticles(ParticleEngine.java:296) at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_59_ParticleEngine_updateParticles_ClientTickEvent.invoke(.dynamic) at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) at net.minecraftforge.fml.common.FMLCommonHandler.onPreClientTick(FMLCommonHandler.java:343)
-- Affected level --Details: Level name: MpServer All players: 1 total; [EntityPlayerSP['Jagereens'/269, l='MpServer', x=-389.95, y=87.98, z=197.79]] Chunk stats: MultiplayerChunkCache: 140, 140 Level seed: 0 Level generator: ID 00 - default, ver 1. Features enabled: false Level generator options:  Level spawn location: World: (32,64,248), Chunk: (at 0,4,8 in 2,15; contains blocks 32,0,240 to 47,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511) Level time: 4478 game time, 4478 day time Level dimension: 0 Level storage version: 0x00000 - Unknown? Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false) Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false Forced entities: 50 total; [EntityWisp['Wisp'/384, l='MpServer', x=-379.16, y=18.57, z=195.84], EntityZombie['Zombie'/270, l='MpServer', x=-398.50, y=29.00, z=199.50], EntityChicken['Chicken'/271, l='MpServer', x=-397.50, y=63.00, z=192.50], EntityZombie['Zombie'/285, l='MpServer', x=-450.50, y=15.00, z=238.50], EntitySkeleton['Skeleton'/286, l='MpServer', x=-450.50, y=58.00, z=239.50], EntityZombie['Zombie'/287, l='MpServer', x=-456.50, y=73.00, z=227.50], EntitySheep['Sheep'/288, l='MpServer', x=-458.55, y=81.00, z=250.45], EntitySheep['Sheep'/289, l='MpServer', x=-455.54, y=82.00, z=252.60], EntitySheep['Sheep'/290, l='MpServer', x=-454.91, y=82.00, z=251.59], EntitySheep['Sheep'/291, l='MpServer', x=-453.50, y=82.00, z=252.50], EntityZombieVillager['Zombie Villager'/302, l='MpServer', x=-445.50, y=23.00, z=190.50], EntityCreeper['Creeper'/303, l='MpServer', x=-416.50, y=36.00, z=160.50], EntitySheep['Sheep'/304, l='MpServer', x=-425.49, y=67.00, z=164.17], EntitySheep['Sheep'/305, l='MpServer', x=-425.40, y=68.00, z=165.55], EntitySheep['Sheep'/306, l='MpServer', x=-425.50, y=72.00, z=171.25], EntitySheep['Sheep'/307, l='MpServer', x=-426.50, y=70.00, z=166.50], EntityZombie['Zombie'/308, l='MpServer', x=-434.50, y=38.00, z=197.50], EntityCreeper['Creeper'/309, l='MpServer', x=-420.50, y=13.00, z=188.50], EntityCreeper['Creeper'/310, l='MpServer', x=-419.50, y=13.00, z=189.50], EntityCreeper['Creeper'/311, l='MpServer', x=-422.50, y=13.00, z=186.50], EntityBat['Bat'/316, l='MpServer', x=-437.25, y=35.49, z=220.75], EntityZombie['Zombie'/317, l='MpServer', x=-446.50, y=58.00, z=238.50], EntityZombie['Zombie'/318, l='MpServer', x=-445.50, y=58.00, z=237.50], EntityPlayerSP['Jagereens'/269, l='MpServer', x=-389.95, y=87.98, z=197.79], EntityMinecartChest['Minecart with Chest'/340, l='MpServer', x=-409.50, y=24.06, z=122.50], EntityBat['Bat'/346, l='MpServer', x=-413.39, y=15.17, z=143.56], EntityBat['Bat'/347, l='MpServer', x=-415.51, y=18.17, z=139.43], EntityChicken['Chicken'/348, l='MpServer', x=-402.50, y=70.00, z=149.50], EntityZombie['Zombie'/349, l='MpServer', x=-389.50, y=27.00, z=130.34], EntityCreeper['Creeper'/351, l='MpServer', x=-407.50, y=17.00, z=173.50], EntityChicken['Chicken'/352, l='MpServer', x=-394.50, y=71.00, z=146.50], EntityCreeper['Creeper'/365, l='MpServer', x=-378.50, y=30.00, z=211.78], EntityItem['item.tile.torch'/366, l='MpServer', x=-383.40, y=27.00, z=141.19], EntityBat['Bat'/367, l='MpServer', x=-407.70, y=30.29, z=189.83], EntityBat['Bat'/368, l='MpServer', x=-413.50, y=31.73, z=190.16], EntityBat['Bat'/369, l='MpServer', x=-411.46, y=33.58, z=183.35], EntityBat['Bat'/370, l='MpServer', x=-382.91, y=23.52, z=154.30], EntityZombie['Zombie'/371, l='MpServer', x=-407.50, y=20.00, z=206.50], EntityZombie['Zombie'/372, l='MpServer', x=-406.50, y=20.00, z=205.50], EntityZombie['Zombie'/373, l='MpServer', x=-408.50, y=20.00, z=206.50], EntityBat['Bat'/374, l='MpServer', x=-412.83, y=30.23, z=194.62], EntityChicken['Chicken'/375, l='MpServer', x=-401.50, y=63.00, z=192.50], EntityChicken['Chicken'/376, l='MpServer', x=-406.50, y=67.00, z=195.50], EntityChicken['Chicken'/377, l='MpServer', x=-404.50, y=65.00, z=193.50], EntityZombie['Zombie'/378, l='MpServer', x=-401.77, y=20.00, z=209.23], EntityZombie['Zombie'/379, l='MpServer', x=-403.72, y=62.00, z=219.51], EntityCreeper['Creeper'/380, l='MpServer', x=-370.50, y=20.00, z=190.50], EntityBat['Bat'/381, l='MpServer', x=-369.20, y=25.11, z=190.68], EntityZombie['Zombie'/382, l='MpServer', x=-392.50, y=20.00, z=216.50], EntitySkeleton['Skeleton'/383, l='MpServer', x=-394.50, y=62.00, z=217.50]] Retry entities: 0 total; [] Server brand: fml,forge Server type: Integrated singleplayer serverStacktrace: at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:420) at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2738) at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:426) at net.minecraft.client.main.Main.main(SourceFile:123) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:483) at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
-- System Details --Details: Minecraft Version: 1.12.2 Operating System: Windows 10 (amd64) version 10.0 Java Version: 1.8.0_25, Oracle Corporation Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation Memory: 151486432 bytes (144 MB) / 671088640 bytes (640 MB) up to 1073741824 bytes (1024 MB) JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=16M IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94 FML: MCP 9.42 Powered by Forge 14.23.3.2702 7 mods loaded, 7 mods active States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
| State     | ID           | Version      | Source                                 | Signature                                | |:--------- |:------------ |:------------ |:-------------------------------------- |:---------------------------------------- | | UCHIJAAAA | minecraft    | 1.12.2       | minecraft.jar                          | None                                     | | UCHIJAAAA | mcp          | 9.42         | minecraft.jar                          | None                                     | | UCHIJAAAA | FML          | 8.0.99.99    | forge-1.12.2-14.23.3.2702.jar          | e3c3d50c7c986df74c645c0ac54639741c90a557 | | UCHIJAAAA | forge        | 14.23.3.2702 | forge-1.12.2-14.23.3.2702.jar          | e3c3d50c7c986df74c645c0ac54639741c90a557 | | UCHIJAAAA | baubles      | 1.5.2        | Baubles-1.12-1.5.2 (1).jar             | None                                     | | UCHIJAAAA | thaumcraft   | 6.1.BETA15   | Thaumcraft-1.12.2-6.1.BETA15.jar       | None                                     | | UCHIJAAAA | xaerominimap | 1.14.9.1     | Xaeros_Minimap_1.14.9.1_Forge_1.12.jar | None                                     |
Loaded coremods (and transformers):  GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 397.93' Renderer: 'GeForce GTX 960M/PCIe/SSE2' Launched Version: 1.12.2-forge1.12.2-14.23.3.2702 LWJGL: 2.9.4 OpenGL: GeForce GTX 960M/PCIe/SSE2 GL version 4.6.0 NVIDIA 397.93, NVIDIA Corporation GL Caps: Using GL 1.3 multitexturing.Using GL 1.3 texture combiners.Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.Shaders are available because OpenGL 2.1 is supported.VBOs are available because OpenGL 1.5 is supported.
Using VBOs: Yes Is Modded: Definitely; Client brand changed to 'fml,forge' Type: Client (map_client.txt) Resource Packs:  Current Language: English (US) Profiler Position: N/A (disabled) CPU: 8x Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz

**Comment by amadeor** [comment]
Can confirm particle-crash at world startup/loading with B15 beta-release. Switching back to B14 solved it for me. Can post crash-logs if anybody wants.

**Comment by Zygus42** [comment]
It's not a crash with world start up etc etc etc.  It's a crash when you move towards or into a Thaumcraft forest biome.  I'm guessing most of the crash reports above are from people who were standing in a Thaumcraft forest before they updated from beta14 to beta15.  Can start worlds fine, but when you fly towards a Thaumcraft forest ....boom to desktop.   And yes, it could well be an entity or somesuch that's spawning in a Thaumcraft Forest biome instead.  

This is a nice clean crash report with no mods other than:
Baubles-1.12-1.5.2
Thaumcraft-1.12.2-6.1.BETA15
forge-1.12.2-14.23.4.2703-installer-win

[crash-2018-05-27_19.53.01-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2042586/crash-2018-05-27_19.53.01-client.txt)

**Comment by NikGold** [comment]
Zygus42 no, youre wrong, ive never found a thamcraft biome in my world.
and nitor wasnt the cause either (had one, rolled back broke and deleted it, tried again same result)
the beta 14 works thought.

**Comment by NikGold** [comment]
Zygus youre wrong, ive never found a thaumcraft forest at all in my world, and nitor isnt the cause either (had only 1, rolled back to beta 14, broke it, deleted it, didnt work either).

**Comment by yunruiwatts** [comment]
@Zygus42 the crash happens when it tries to render any Thaumcraft particle. Magical forests are one source of such particles, but using salis mundus and nitor are too. So pretty much everyone is affected unless they avoid everything added by this mod.

For everyone: normally it is helpful for people to post more information, but in this case the bug is very clear - Azanor tried to use a `CopyOnWriteArrayList` to fix an issue, but that type of list throws an error when you try to use one of the methods that the code had been using. I'm not surprised he didn't notice that caveat beforehand - it was not trivial to find in the documentation.

TL;DR - Azanor doesn't need more information to be able to fix the issue.

**Comment by enderger** [comment]
How would the magical forest biome effect me in the middle of the aether demension?

**Comment by Sanofus** [comment]
I had that issue deep at the bed rock without anything from thaumcraft. And I could solve it just downgrading to beta 14. It happens to me only in Beta 15.

**Comment by Foxgreat** [comment]
It's some like particle from monster or any else, causes effect of someones entity, in case of flatland crashing on same zone.

And I think its enough for azanor to fix that)

**Comment by enderger** [comment]
Looking at my B14 crash and the current B15 crash, the concurrentException is causing my crash and the new crash is caused by the copyOnWriteArrayList. It seems that both throw an exception, so we need another type of list.

**Comment by enderger** [comment]
In a way, the bug was amplified by the fix.

**Comment by bdudek** [comment]
im getting the same crash in b15 while just wondering the nether https://pastebin.com/aP5huBuB

**Comment by Geethebluesky** [comment]
I wonder, are there wisps in the Nether? They're a source of particles, right? 

(I haven't seen Nether wisps in my playthrough, have lots of other mobs in the spawn table making them rarer.)

**Comment by bdudek** [comment]
i have seen wisps in earlier versions before in the nether

**Comment by ireece1** [comment]
Yeah i have this same issue after updating to Beta 15.

**Comment by Sanofus** [comment]
> I wonder, are there wisps in the Nether? They're a source of particles, right?

> (I haven't seen Nether wisps in my playthrough, have lots of other mobs in the spawn table making them rarer.)

I haven't seen any Wisps in the nether.

**Comment by zealwill** [comment]
updated from beta 13 to 15, crash on load of single player world with thaumcraft research. (log 1)
no crash generating a new world and running around for a few minuets. 
opened previously generated world no TC research no crash. JEI acquired yellow nitor placed and caused crash (log 2)

[crash-2018-05-27_16.29.55-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2043000/crash-2018-05-27_16.29.55-client.txt)
[crash-2018-05-27_16.44.33-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2043001/crash-2018-05-27_16.44.33-client.txt)

**Comment by waddon1** [comment]
Also just experienced this crash 
[crash-2018-05-28_00.06.06-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2043120/crash-2018-05-28_00.06.06-client.txt)

**Comment by Geethebluesky** [comment]
Try reverting back to Beta 14 for now (Or beta 13 if 14 doesn't play nice for you), there won't be a solution until a fix is released.

**Comment by enderger** [comment]
Can someone pin this issue

**Comment by wamc2017** [comment]
There's no possibility to pin issues, as far as I know and if so, Azanor is the only one who has rights on this tracker...

**Comment by enderger** [comment]
We need a mod for this thread to pin issues that are important

**Comment by MokahTGS** [comment]
Stop asking for this to be pinned.  All issues are important and this has been reported.  Azanor will do what he feels needs to be done.  Constantly posting with no new or helpful information is useless and annoying.

**Comment by enderger** [comment]
For future commenters...
Please put your crash reports in https://gist.github.com, https://www.pastebin.com, or something similar.

**Comment by Notawallplan** [comment]
Can this issue be avoided by disabling shaders from the config?

**Comment by Sanofus** [comment]
No, it cannot.

**Comment by enderger** [comment]
Does this mean B16 is coming out soon?

**Comment by Azanor** [comment]
It is already out.

**Comment by GeorgeWL** [comment]
@jagereens next time, please copy it into a [gist.github.com](gist.github.com). That's a very big wall of text and makes it difficult to read the rest of the comments on some screens.



_Sent from my Pixel 2 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
